apps/projects/contracts/Projects.sol

```diff
@@ -386,39 +371,47 @@ contract Projects is IsContract, AragonApp {
      * @param _requestor address of user that will be assigned the issue
      */
     function approveAssignment(
-        bytes32 _repoId, 
-        uint256 _issueNumber, 
-        address _requestor
+        bytes32 _repoId,
+        uint256 _issueNumber,
+        address _requestor,
+        string _updatedApplication
     ) external isInitialized auth(TASK_ASSIGNMENT_ROLE)
     {
         require(bytes(repos[_repoId].issues[_issueNumber].assignmentRequests[_requestor]).length != 0, "User has not applied for this issue");
         repos[_repoId].issues[_issueNumber].assignee = _requestor;
+        repos[_repoId].issues[_issueNumber].assignmentRequests[_requestor] = _updatedApplication;
 
         emit AssignmentApproved(_requestor, _repoId, _issueNumber);
     }
 
     /**
-     * @notice Submit work for issue '`_issueNumber`'.
+     * @notice Submit work for issue `_issueNumber`.
      * @dev add a submission to local state after it's been added to StandardBounties.sol
      * @param _repoId the github repo id of the issue
      * @param _issueNumber the github issue up for assignment
      * @param _submissionAddress IPFS hash of the Pull Request
-     * //param _fulfillmentId retrieved from event after the work is submitted to the bounties contract externally 
+     * //param _fulfillmentId retrieved from event after the work is submitted to the bounties contract externally
      */
     function submitWork(
-        bytes32 _repoId, 
-        uint256 _issueNumber, 
+        bytes32 _repoId,
+        uint256 _issueNumber,
         string _submissionAddress
         //uint256 _fulfillmentId
     ) external isInitialized
     {
-        require(msg.sender == repos[_repoId].issues[_issueNumber].assignee, "User not assigned to this issue");
         GithubIssue storage issue = repos[_repoId].issues[_issueNumber];
+        require(!issue.fulfilled,"BOUNTY_FULFILLED");
+        require(msg.sender == issue.assignee, "USER_NOT_ASSIGNED");
         bounties.fulfillBounty(issue.standardBountyId, _submissionAddress);
-        issue.workSubmissions[msg.sender] = WorkSubmission(
-            SubmissionStatus.Unreviewed,
-            _submissionAddress,
-            issue.submissionQty
+        issue.submissionIndices.push(
+            workSubmissions.push(
+                WorkSubmission(
+                    SubmissionStatus.Unreviewed,
+                    _submissionAddress,
+                    issue.submissionQty,
+                    issue.assignee
+                )
+            ) - 1 // push returns array length so we need to subtract 1 to get the index value
```

Comment:
Should we get rid of issue.submissionQty since we can rely into issue.submissionIndices.length ?  (line 417)

Reply:
Done